library/std/src/collections/mod.rs

```diff
@@ -401,9 +401,10 @@
 #![stable(feature = "rust1", since = "1.0.0")]
 
 #[stable(feature = "rust1", since = "1.0.0")]
-#[rustc_deprecated(reason = "moved to `std::ops::Bound`", since = "1.26.0")]
+#[rustc_deprecated(reason = "moved to `std::ops::Bound`", since = "1.52.0")]
 #[doc(hidden)]
-pub use crate::ops::Bound;
+pub type Bound<T> = crate::ops::Bound<T>;
```

Comment:
FWIW, this is technically a breaking change if someone did `use std::collections::Bound::*;` (like Miri did)

Reply:
Oh, I assumed miri had broken due to denying deprecation warnings. I wasn't aware that type aliases to enums couldn't glob-import; it looks like even concrete importing like `use collections::Bound::Unbounded` isn't supported. :\ This is a distressing limitation of type alises that I wasn't aware of; I trusted that the limitations laid out in the reference were exhaustive. I'm not sure what to do about this. At the very least I'll file an issue about supporting this, since it really does seem like it ought to be supported.

Reply:
I've filed https://github.com/rust-lang/rust/issues/83248 to track importing variants through type aliases. I'll also ping the libs team on Zulip to see what they think should be done.

Reply:
Zulip thread at https://rust-lang.zulipchat.com/#narrow/stream/219381-t-libs/topic/unavoidable.20breakage.20when.20deprecating.20an.20enum.3F .

Reply:
>  I trusted that the limitations laid out in the reference were exhaustive.

Which part of the reference is that? Might be worth reporting a bug (or submitting a PR).

Reply:
This page of the reference: https://doc.rust-lang.org/reference/items/type-aliases.html , at the bottom where it talks about where type aliases cannot be used. I'll submit a PR.

Reply:
Reference PR at https://github.com/rust-lang/reference/pull/984